src/note_compiler.h

```diff
@@ -22,22 +22,30 @@ typedef enum NoteCompilerState {
 	NUM_NOTE_COMPILER_STATES,
 } NoteCompilerState;
 
+typedef enum NoteCompilerErrorState {
+	NOTE_COMPILER_ERROR_STATE_UNKNOWN_ERROR = 0,
+	NOTE_COMPILER_ERROR_STATE_UNEXPECTED_CHARACTER,
+} NoteCompilerErrorState;
+
 struct NoteCompiler {
-	char* bar;
-	NoteCompilerState state;
-	size_t bar_length;
+	char* filename;
```

Comment:
If we pass the filename in the constructor here, we should probably do that for all the other modules as well.  
(Maybe not for the interpreter, if the idea is to be able to potentially load modules from different files?)

Also need to think about scope. (Caller needs to guarantee the string stays around for as long as the compiler/lexer whatever is used. This is fine, but should really be documented.)

Reply:
(Thinking more about this, we might want to store the (pointer-to) filename in the tokens instead. There's little sense in having the line information in one place, and the file name in another.)